packages/react-form-state/src/FormState.ts

```diff
@@ -212,21 +207,31 @@ export default class FormState<
     } else {
       this.setState({submitting: false, errors});
     }
-  }
+  };
 
-  @memoize()
-  @bind()
-  private fieldWithHandlers<Key extends keyof Fields>(
+  private fieldWithHandlers = <Key extends keyof Fields>(
     field: FieldStates<Fields>[Key],
-    fieldPath: Key,
-  ) {
-    return {
-      ...(field as FieldState<Fields[Key]>),
-      name: fieldPath,
-      onChange: this.updateField.bind(this, fieldPath),
-      onBlur: this.blurField.bind(this, fieldPath),
+    fieldPath: string & Key,
```

Comment:
why?

Reply:
Typescript didn't like it without this. I forgot to investigate after getting the build to work; thanks for catching.